stratosphere/pm/source/impl/pm_resource_manager.cpp

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (c) 2018-2019 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stratosphere/spl.hpp>
+
+#include "pm_resource_manager.hpp"
+
+namespace sts::pm::resource {
+
+    namespace {
+
+        constexpr LimitableResource LimitableResources[] = {
+            LimitableResource_Memory,
+            LimitableResource_Threads,
+            LimitableResource_Events,
+            LimitableResource_TransferMemories,
+            LimitableResource_Sessions,
+        };
+        constexpr size_t LimitableResource_Count = sizeof(LimitableResources) / sizeof(LimitableResources[0]);
+
+        constexpr size_t Megabyte = 0x100000;
```

Comment:
Might be worth using [user-defined literals](https://en.cppreference.com/w/cpp/language/user_literal) but that might be slight overkill and I can't think of a suitable namespace in which to place its definition.

Reply:
Yeah, seems a bit overkill.